clients/admin-ui/cypress/e2e/datasets.cy.ts

```diff
@@ -234,6 +234,163 @@ describe("Dataset", () => {
+        // type just one space character to make sure the text field triggers Formik's handlers
+        cy.getByTestId("input-datasetYaml").type(" ");
+
+        cy.getByTestId("create-dataset-btn").click();
+        cy.wait("@postDataset").then((interception) => {
+          const { body } = interception.request;
+          expect(body).to.eql(dataset);
+        });
+
+        // should navigate to the created dataset
+        cy.getByTestId("toast-success-msg");
+        cy.url().should("contain", `dataset/${key}`);
+      });
+    });
+
+    it("Can render errors in yaml", () => {
+      cy.intercept("POST", "/api/v1/dataset", {
+        statusCode: 422,
+        body: {
+          detail: [
+            {
+              loc: ["body", "fides_key"],
+              msg: "field required",
+              type: "value_error.missing",
+            },
+            {
+              loc: ["body", "collections"],
+              msg: "field required",
+              type: "value_error.missing",
+            },
+          ],
+        },
+      }).as("postDataset");
+      cy.visit("/dataset/new");
+      cy.getByTestId("upload-yaml-btn").click();
+      // type something that isn't yaml
+      cy.getByTestId("input-datasetYaml").type("invalid: invalid: invalid");
+      cy.getByTestId("create-dataset-btn").click();
+      cy.getByTestId("error-datasetYaml").should("contain", "Could not parse");
+
+      // type something that is valid yaml and let backend render an error
+      cy.getByTestId("input-datasetYaml")
+        .clear()
+        .type("valid yaml that is not a dataset");
+      cy.getByTestId("create-dataset-btn").click();
+      cy.getByTestId("error-datasetYaml").should("contain", "field required");
+    });
+
+    it("Can create a dataset by connecting to a database", () => {
+      const connectionString =
+        "postgresql://postgres:fidesctl@fidesctl-db:5432/fidesctl_test";
+      cy.intercept("POST", "/api/v1/generate", {
+        fixture: "generate_dataset.json",
+      }).as("postGenerate");
+      cy.intercept("POST", "/api/v1/dataset", { fixture: "dataset.json" }).as(
+        "postDataset"
+      );
+      cy.visit("/dataset/new");
+      cy.getByTestId("connect-db-btn").click();
+      cy.getByTestId("input-url").type(connectionString);
+      cy.getByTestId("create-dataset-btn").click();
+      cy.wait("@postGenerate").then((interception) => {
+        expect(
+          interception.request.body.generate.config.connection_string
+        ).to.eql(connectionString);
+      });
+      cy.wait("@postDataset").then((interception) => {
+        // should be whatever is in the generate fixture
+        expect(interception.request.body.fides_key).to.eql("public");
+      });
+      // should navigate to the created dataset
+      cy.getByTestId("toast-success-msg");
+      cy.url().should("contain", `dataset/demo_users_dataset`);
+    });
+
+    it.only("Can render errors when connecting to a database", () => {
```

Comment:
Don't want to check in `it.only`. I'm surprised this isn't part of cypress's eslint by default, but I found [a package](https://www.npmjs.com/package/eslint-plugin-no-only-tests) for it.